app/src/main/java/com/vero/tasky/agenda/data/worker/UpdateEventWorker.kt

```diff
@@ -0,0 +1,77 @@
+package com.vero.tasky.agenda.data.worker
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.squareup.moshi.Moshi
+import com.vero.tasky.agenda.data.local.dao.EventDao
+import com.vero.tasky.agenda.data.local.dao.ModifiedAgendaItemDao
+import com.vero.tasky.agenda.data.local.entities.ModifiedAgendaItemEntity
+import com.vero.tasky.agenda.data.mappers.toLocalPhoto
+import com.vero.tasky.agenda.data.mappers.toUpdateEventRequest
+import com.vero.tasky.agenda.data.remote.network.api.EventApi
+import com.vero.tasky.agenda.data.remote.network.request.UpdateEventRequest
+import com.vero.tasky.agenda.data.util.multipart.MultipartParser
+import com.vero.tasky.agenda.domain.model.AgendaItemType
+import com.vero.tasky.agenda.domain.model.ModificationType
+import com.vero.tasky.agenda.domain.workmanagerrunner.UpdateEventWorkerRunner
+import com.vero.tasky.core.data.remote.safeSuspendCall
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import okhttp3.MultipartBody
+
+@HiltWorker
+class UpdateEventWorker @AssistedInject constructor(
```

Comment:
Since all the logic is pretty much similar, you can have it in 1 worker, and change api call and adapter (mentioned on another comment)